src/IO/H5/File.hpp

```diff
@@ -178,6 +178,9 @@ class H5File {
   hid_t file_id_{-1};
   mutable std::unique_ptr<h5::Object> current_object_{nullptr};
   std::vector<std::string> h5_groups_;
+  // used to prevent the destructor from throwing an exception for
```

Comment:
Destructors should never throw anyway, so we need some other way of handling this. I don't remember exactly, but I think that's an explicit rule in C++. You at least get really bad behavior if a destructor is ever allowed to throw. I'd say we just try to close the file and maybe just print a message if we can't for some reason.